internal/pkg/config/v1/context.go

```diff
@@ -191,32 +198,33 @@ func (c *Context) GetSuspensionStatus() *ccloudv1.SuspensionStatus {
 	return c.GetOrganization().GetSuspensionStatus()
 }
 
-func (c *Context) GetEnvironment() *ccloudv1.Account {
-	if auth := c.GetAuth(); auth != nil {
-		return auth.Account
+func (c *Context) GetCurrentEnvironment() string {
+	if c.CurrentEnvironment != "" {
+		return c.CurrentEnvironment
 	}
-	return nil
-}
 
-func (c *Context) SetEnvironment(environment *ccloudv1.Account) {
-	if c.GetAuth() == nil {
-		c.SetAuth(new(AuthConfig))
+	if auth := c.GetAuth(); auth != nil {
+		return auth.Account.GetId()
 	}
-	c.GetAuth().Account = environment
+
+	return ""
 }
 
-func (c *Context) GetEnvironments() []*ccloudv1.Account {
+func (c *Context) SetCurrentEnvironment(id string) {
+	c.CurrentEnvironment = id
+
 	if auth := c.GetAuth(); auth != nil {
-		return auth.Accounts
+		auth.Account = nil
+		auth.Accounts = nil
 	}
-	return nil
-}
 
-func (c *Context) SetEnvironments(environments []*ccloudv1.Account) {
-	if c.GetAuth() == nil {
-		c.SetAuth(new(AuthConfig))
+	if id != "" {
+		if _, ok := c.Environments[id]; !ok {
+			c.Environments[id] = &EnvironmentContext{}
+		}
+	} else {
+		delete(c.Environments, id)
```

Comment:
I'm confused about this line. Do we expect/suspect that `c.Environments` has an empty key if this function is called with an empty `id`?

Reply:
Oops, this should be `c.CurrentEnvironment`, not `id`.

Reply:
Ah, good point. I don't think this is right.

Reply:
Ah, that makes sense!